pyincore/semanticservice.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) 2019 University of Illinois and others. All rights reserved.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Mozilla Public License v2.0 which accompanies this distribution,
+# and is available at https://www.mozilla.org/en-US/MPL/2.0/
+
+
+import json
+import os
+from typing import Tuple, Union
+
+import pyincore.globals as pyglobals
+from pyincore import IncoreClient
+from urllib.parse import urljoin
+import requests
+
+logger = pyglobals.LOGGER
+
+
+class SemanticService:
+    """Semantics Service client.
+
+    Use this class to interact with the semantics service.
+
+    Args:
+        client (IncoreClient): Service authentication.
+
+    """
+
+    def __init__(self, client: IncoreClient):
+        self.client = client
+        self.base_url = urljoin(client.service_url, "semantics/api/types")
+
+    @staticmethod
+    def return_http_response(http_response):
```

Comment:
Not related to this task, i notice this function has been repetitively exist for every service; anyway we can place them in a centralized place?

Reply:
hahah, I was starting to think that too

Reply:
where do you suggest?

Reply:
client.py? Just random thoughts i haven't think through it yet.

Reply:
Lets resolve this in a separate issue.